stdlib/public/core/Collection.swift

```diff
@@ -1325,13 +1325,15 @@ extension Collection {
       return []
     }
 
-    var result = ContiguousArray<T>()
-    result.reserveCapacity(n)
+    let result: ContiguousArray<T>
+    var p: UnsafeMutablePointer<T>
+    (result, p) = ContiguousArray<T>._allocateUninitialized(count)
```

Comment:
Can you switch `count` to `n` here, in case a collection is super broken and returns a different count here compared to up above? Otherwise we might write past the buffer in the loop below.